### PR TITLE
[Merged by Bors] - feat: Support podOverrides

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [x] (Integration-)Test cases added
- [ ] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

**Comment by sbernauer** [comment]
Started https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/nifi-operator-it-custom/35/

File: tests/templates/kuttl/resources/02-install-nifi.yaml.j2 [hunk 1]
```diff
@@ -52,6 +55,24 @@ spec:
...
+              min: 600m
+              max: 1100m
+            memory:
+              limit: 3Gi
+        replicas: 1
+      resources-from-pod-overrides:
+        podOverrides:
+          spec:
+            containers:
+              - name: hbase
```
Comment (razvan) [inline tests/templates/kuttl/resources/02-install-nifi.yaml.j2:73]:
```suggestion
              - name: nifi
```

File: tests/templates/kuttl/resources/02-assert.yaml.2 [hunk 2]
```diff
@@ -6,23 +6,68 @@ timeout: 600
...
-          memory: 2Gi
-        requests:
-          memory: 2Gi
+  template:
+    spec:
+      containers:
+        - name: nifi
+          resources:
+            requests:
+              cpu: 600m
```
Comment (razvan) [inline tests/templates/kuttl/resources/02-assert.yaml.2:63]: shouldn't this be 700m? and the limit 1200m?

**Review by razvan — COMMENTED** [review]
code looks good but I'm not sure about the tests

File: tests/templates/kuttl/resources/02-assert.yaml.2 (hunk 2 above)
Comment (sbernauer) [reply tests/templates/kuttl/resources/02-assert.yaml.2:63]: Sure thing, was a leftover

Comment (sbernauer) [reply tests/templates/kuttl/resources/02-assert.yaml.2:63]: Lol this had no effect as the file ended in `.2` instead of `.j2` :shrug:

**Comment by sbernauer** [comment]
resources test passed locally

**Review by razvan — APPROVED** [review]

**Comment by sbernauer** [comment]
bors r+
Thx!

**Pull request closed without merging**